### PR TITLE
Fix python operator with side effects.

### PR DESCRIPTION
**Pull request by banasraf**
Signed-off-by: Rafal <Banas.Rafal97@gmail.com>

#### Why we need this PR?
- It fixes a race condition in the python operator with side effects.

#### What happened in this PR?
 - Explain solution of the problem, new feature added.
    A global mutex was added in python operator implementation to protect the python function
    execution.
 - What was changed, added, removed?
    A simple change in _python_function.cc_
 - Was this PR tested? How?
    A test in _test\_operator\_python\_function.py_ was added.

**JIRA TASK**: [DALI-837]

### CODE REVIEW

**Comment by banasraf** [comment]
!build

File: dali/test/python/test_operator_python_function.py [hunk 1]
```diff
@@ -354,3 +354,39 @@ def test_sink():
...
+        out_two, = pipe_two.run()
+
+        print('Iter ' + str(it) + ' Len one ' + str(len(out_one)) + ' len two ' + str(len(out_two)))
+        assert counter == len(out_one) + len(out_two)
+
+        for s in range(BATCH_SIZE):
+            elem_one = out_one.at(s)[0][0][0]
+            elem_two = out_two.at(s)[0][0][0]
+
+            assert 0 < elem_one <= len(out_one)
```
Comment (JanuszL) [inline dali/test/python/test_operator_python_function.py]: Can we put more strict constrains or there is no predictable relation between sample index and value inside it?

Comment (banasraf) [reply dali/test/python/test_operator_python_function.py]: I don't see any more strict constrain as we can't make any assumptions about the order of sample processing. But it is quite good test, as usually in a race condition the counter ends up being too low and it's caught by the constrain for the second pipeline's result.

Comment (awolant) [reply dali/test/python/test_operator_python_function.py]: Maybe we could sort and check if everything is there, as you were doing it locally?

Comment (banasraf) [reply dali/test/python/test_operator_python_function.py]: Yeah it sounds better. Done.

**Comment by banasraf** [comment]
!build

**Comment by awolant** [comment]
!build

**Pull request merged**